packages/client/hmi-client/src/components/dataset/tera-dataset-jupyter-panel.vue

```diff
@@ -182,13 +182,24 @@ const setKernelContext = (kernel: IKernelConnection, context_info) => {
 	kernel?.sendJupyterMessage(message);
 };
 
+// FIXME: this is a bit fragile
+const toAssetType = (t: string) => {
+	if (t.endsWith('Id')) {
+		return t.substring(0, t.length - 2);
+	}
+	throw new Error(`Cannot convert type ${t}`);
+};
+
```

Comment:
Here a stronger method to put your mind at ease.

```suggestion
function toAssetType(type: string): AssetType {
    const lowercaseType = type.toLowerCase();

    for (const assetType of Object.values(AssetType)) {
        if (lowercaseType.includes(assetType.toLowerCase())) {
            return assetType;
        }
    }

    throw new Error(`Cannot convert ${type} to a known AssetType.`);
}
```

When tested with the following it works as you expect it:
```js
console.log(toAssetType('modelID'));
console.log(toAssetType('lolDataset'));
console.log(toAssetType('I love using beaker to analyse my simulations results'));
console.log(toAssetType('image'));
```

![Screenshot 2024-02-23 at 08 16 43](https://github.com/DARPA-ASKEM/terarium/assets/636801/264eac5d-4d5a-4cb0-a52d-53c7e14d3f23)

Reply:
Thanks for the offer, but no it does not. 
This is meant to convert the string to match askem-beaker's terminology, and and the terms used in Terarium. I will add a bit more explanation comment on why the function is a bit fragile.